src/Symfony/Component/HttpClient/Response/JsonMockResponse.php

```diff
@@ -16,18 +16,35 @@
 class JsonMockResponse extends MockResponse
 {
     /**
-     * @param mixed $body Any value that `json_encode()` can serialize
+     * @param mixed $data Any value that `json_encode()` can serialize
+     * @param bool  $json Whether the data is already a JSON string
      */
-    public function __construct(mixed $body = [], array $info = [])
+    public function __construct(mixed $data = [], array $info = [], bool $json = false)
```

Comment:
```suggestion
    public function __construct(mixed $data = [], array $info = [], bool $isJson = false)
```
to be more speakable

Reply:
Looking at this again, I think it's acceptable to json_decode and then json_encode for simplicity instead of having to add a new arg etc. The perf gain is minimal I guess?

Reply:
Thanks